File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    answer = [0] * len(id_list)
    divReport = []
    checkReport = [0] * len(id_list)
    stopUser = []
    dic = {}
    count = 0
    
    emp = set(report)
    report = list(emp)
    
    for i in range(len(id_list)):
        dic[id_list[i]] = []
    
    for i in range(len(report)):
        divReport.append(report[i].split(' '))
    
    for i in range(len(divReport)):
        dic[divReport[i][1]].append(divReport[i][0])
    
    for i in dic:
        if len(dic[i]) >= k :
            stopUser.append(i)
    
    for i in stopUser:
        for j in dic[i]:
            answer[id_list.index(j)] += 1
    
    return answer

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        
        HashSet<String> set = new HashSet<>(Arrays.asList(report));
        ArrayList<String> list = new ArrayList<>(set);
        HashMap<String, Integer> complain = new HashMap<>();
        ArrayList<String> result = new ArrayList<>();
        
        for(int i = 0; i < list.size(); i++){
            String[] str = list.get(i).split(" ");
            if (!complain.containsKey(str[1])) {
                complain.put(str[1], 1);
            } else {
                complain.put(str[1], complain.get(str[1]) + 1);
            }
        }
        
        for(int i = 0; i < id_list.length; i++){
            result.add(id_list[i]);
        }
        
        for(int i = 0; i < list.size(); i++){
            String[] str = list.get(i).split(" ");
            if (complain.get(str[1]) >= k) {
                answer[result.indexOf(str[0])] += 1;
            }
        }
        
        return answer;
    }
}